overlord/devicestate/devicestate.go

```diff
@@ -490,9 +532,9 @@ func remodelEssentialSnapTasks(ctx context.Context, st *state.State, ms modelSna
 			return nil, err
 		}
 		if changed {
-			ts, err := snapstateUpdateWithDeviceContext(st, ms.newSnap,
-				&snapstate.RevisionOptions{Channel: newModelSnapChannel},
-				userID, snapstate.Flags{NoReRefresh: true}, deviceCtx, fromChange)
+			ts, err := remodelOpts.UpdateWithDeviceContext(st,
+				localSi, path, ms.newSnap, newModelSnapChannel, userID,
+				snapstate.Flags{NoReRefresh: true}, deviceCtx, fromChange)
```

Comment:
we also seem not reach here with the need for local snaps

Reply:
Added tests for this